### DISCUSSION TITLE
Nextjs and Axios

### DISCUSSION THREAD

**Discussion by fouadhallal** [vercel/next.js#65042] [Help]
### Summary

I am new to nextjs and was looking for examples/template on using Axios messaging with Next.js.  ANy info is much appreciated.  Also, is Axios a reliable service? any other options?

Thank you,
Fouad

**Comment by shanakasp** [comment]
Next.js is a versatile framework for building React applications, and Axios is a popular choice for making HTTP requests within those applications due to its simplicity and flexibility. For installing axios npm install axios.  Also the code looks like this, // pages/api/data.js
import axios from 'axios';

This is and example API route for the Server side, 

export default async (req, res) => {
  try {
    const response = await axios.get('https://api.example.com/data');
    res.status(200).json(response.data);
  } catch (error) {
    console.error('Error fetching data:', error);
    res.status(500).json({ error: 'Internal Server Error' });
  }
};

And lets see how to use this in client side, 

import { useState, useEffect } from 'react';
import axios from 'axios';

const MyComponent = () => {
  const [data, setData] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      try {
        const response = await axios.get('/api/data');
        setData(response.data);
      } catch (error) {
        console.error('Error fetching data:', error);
      }
    };

    fetchData();
  }, []);

  return (
    <div>
      {data ? (
        <pre>{JSON.stringify(data, null, 2)}</pre>
      ) : (
        <p>Loading...</p>
      )}
    </div>
  );
};

export default MyComponent;


Also I will suggest few tutorials to get clear understanding about this, I think netNinja crash course better for beginner. https://www.youtube.com/watch?v=TJQbDPGzm0Y&list=PL4cUxeGkcC9jZIVqmy_QhfQdi6mzQvJnT

**Reply by fouadhallal** [reply]
Thank you. This is very helpful
